plugins/preview/plugin.js

```diff
@@ -8,128 +8,262 @@
+				previewLocation = getPreviewLocation(),
+				previewUrl = getPreviewUrl(),
+				previewWindow,
+				doc;
 
 			// (https://dev.ckeditor.com/ticket/9907) Allow data manipulation before preview is displayed.
 			// Also don't open the preview window when event cancelled.
-			if ( editor.fire( 'contentPreview', eventData = { dataValue: sHTML } ) === false )
+			if ( editor.fire( 'contentPreview', eventData ) === false ) {
 				return false;
-
-			var sOpenUrl = '',
-				ieLocation;
-
-			if ( CKEDITOR.env.ie ) {
-				window._cke_htmlToLoad = eventData.dataValue;
-				ieLocation = 'javascript:void( (function(){' + // jshint ignore:line
-					'document.open();' +
-					// Support for custom document.domain.
-					// Strip comments and replace parent with window.opener in the function body.
-					( '(' + CKEDITOR.tools.fixDomain + ')();' ).replace( /\/\/.*?\n/g, '' ).replace( /parent\./g, 'window.opener.' ) +
-					'document.write( window.opener._cke_htmlToLoad );' +
-					'document.close();' +
-					'window.opener._cke_htmlToLoad = null;' +
-				'})() )';
-				// For IE we should use window.location rather than setting url in window.open. (https://dev.ckeditor.com/ticket/11146)
-				sOpenUrl = '';
 			}
 
-			// With Firefox only, we need to open a special preview page, so
-			// anchors will work properly on it. (https://dev.ckeditor.com/ticket/9047)
-			if ( CKEDITOR.env.gecko ) {
+			// In cases where we force reloading the location or setting concrete URL to open,
+			// we need a way to pass content to the opened window. We do it by hack with
+			// passing it through window's parent property.
+			if ( previewLocation || previewUrl ) {
 				window._cke_htmlToLoad = eventData.dataValue;
-				sOpenUrl = CKEDITOR.getUrl( pluginPath + 'preview.html' );
 			}
 
-			var oWindow = window.open( sOpenUrl, null, 'toolbar=yes,location=no,status=yes,menubar=yes,scrollbars=yes,resizable=yes,width=' +
-				iWidth + ',height=' + iHeight + ',left=' + iLeft );
+			previewWindow = window.open( previewUrl, null, generateWindowOptions( windowDimensions ) );
+
+			// For IE we want to assign whole js stored in previewLocation, but in case of
+			// popup blocker activation oWindow variable will be null (https://dev.ckeditor.com/ticket/11597).
+			if ( previewLocation && previewWindow ) {
+				previewWindow.location = previewLocation;
+			}
 
-			// For IE we want to assign whole js stored in ieLocation, but in case of
-			// popup blocker activation oWindow variable will be null. (https://dev.ckeditor.com/ticket/11597)
-			if ( CKEDITOR.env.ie && oWindow )
-				oWindow.location = ieLocation;
+			if ( !window._cke_htmlToLoad ) {
+				doc = previewWindow.document;
 
-			if ( !CKEDITOR.env.ie && !CKEDITOR.env.gecko ) {
-				var doc = oWindow.document;
 				doc.open();
 				doc.write( eventData.dataValue );
 				doc.close();
 			}
 
-			return true;
+			return new CKEDITOR.dom.window( previewWindow );
 		}
 	};
 
-	var pluginName = 'preview';
+	previewCmd = {
```

Comment:
Defining the command definition separately doesn't seem important. Could it be inlined when adding command in `plugin#init` method?